platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCache.kt

```diff
@@ -154,6 +155,9 @@ class DefaultConfigurationCache internal constructor(
     val gradlePropertiesController: GradlePropertiesController
         get() = host.service()
 
+    private
+    val currentCandidateEntries = mutableListOf<CandidateEntry>()
```

Comment:
❌ I'd model the state transitions more carefully here. Perhaps it could be nullable and initially `null` to signal a `non loaded state`, or even something more explicit like a sealed class hierarchy.

Reply:
@bamboo Modelled it as a nullable list.